modules/afsmtp/afsmtp.c

```diff
@@ -246,18 +243,38 @@ afsmtp_dd_msg_add_recipient(AFSMTPRecipient *rcpt, smtp_message_t message)
     default:
       return;
     }
-  smtp_set_header(message, hdr, rcpt->phrase, rcpt->address);
+
+  smtp_set_header(message, hdr, rcpt->phrase, afsmtp_wash_string (self->str->str));
   smtp_set_header_option(message, hdr, Hdr_OVERRIDE, 1);
 }
 
 static void
+afsmtp_dd_msg_add_recipient_with_template(smtp_message_t message, AFSMTPDriver *self, LogTemplate *addr_tmpl, LogMessage *msg)
```

Comment:
This function should be renamed to `_smtp_message_add_recipient_from_template()`.